src/apps/replication/lib/replica_learn.cpp

```diff
@@ -212,7 +214,7 @@ void replica::on_learn(dsn_message_t msg, const learn_request& request)
     auto it = _primary_states.learners.find(request.learner);
     if (it == _primary_states.learners.end())
     {
-        response.err = (response.config.status == PS_SECONDARY ? ERR_OK : ERR_OBJECT_NOT_FOUND);
+        response.err = ERR_OBJECT_NOT_FOUND;
```

Comment:
when the learner is already a secondary, we should return ERR_OK so that on_learn_reply will update the state to PS_SECONDARY on learner.